README.md

```diff
@@ -92,145 +100,77 @@ In this project we focused on integrating directly with Maven + NAR.
-test harness.  A test harness includes code for mocks, test drivers, and any
+test harness. A test harness includes code for mocks, test drivers, and any
 setup that needs to be done before performing the actual test.
 
 There are two ways to do this:
-1. Add new code to the git repo.  If you can't add code to the central repo, a
-   common approach is to maintain a fork with your local additions. We took
-   this approach.
-2. Create a new repository with just the harnesses. This typically works ok for
-   libraries, but is less generic.  It is generally only used when the team
-   writing the code (e.g., the security team) has little interaction with the
-   developers and wants to keep all their tests completely separate.
+1. Create a new Maven project with just the harnesses that import the system under
+  test via Maven dependency management. This typically works great for libraries where
+  the harness can use the unmodified objects being built as-is.
+2. Sometimes, code you want to test is contained in an executable (not a library). In 
+  this case, new harness code would need to be added directly to the Maven module that contains 
+  the system under test. If you can't add code to the central repo, a common approach is to 
+  maintain a fork with your local additions. We took this approach.
 
+In this case, the interface the library exports suffices and we'll do (1).
 
-In this case we'll do (1).
-
-### Step 3: Modify the `pom.xml` file to include your new code.
+### Step 3: Create a `myharness` module.
 
 Maven has a very opinionated default directory structure.  It expects code in
 `src/main/cpp` for C++ code, includes in `src/main/include`, and tests in
 `src/test/cpp`.  While you can override these, it's often easier just to work
 with defaults.  In our case, we're going to add the harness code to the default
-`src/test/cpp`.
- 
-The tricky part was figuring out how to modify `pom.xml`, as Maven + NAR has
-a far smaller community to learn from.  We finally found a [great
-writeup](https://groups.google.com/g/maven-nar/c/-XSwh3l47Ow) and example from
-Jef Douglas on the NAR mailing list.  He even put together a [Github
-repo](https://github.com/dugilos/nar-test) demonstrating the idea.
-
-The key part for us was to understand that the test name in the `pom.xml` file
-are related to how you name your tests.  If you have the test named `test1`, by
-default this will refer to `src/test/cpp/test1.cpp`.  
-
-Jef gives a wonderful example. Suppose you have the configuration:
-```xml
-<configuration>
-
-    <tests>
-
-        <test>
-            <name>test1</name>
-            <link>shared</link>
-            <run>true</run>
-        </test>
-
-        <test>
-            <name>test2</name>
-            <link>shared</link>
-            <run>true</run>
-            <args>
-                <arg>param1</arg>
-                <arg>param2</arg>
-            </args>
-        </test>
-
-    </tests>
-
-</configuration>
-```
```

Comment:
I hate to lose this knowledge, and on reflection, I can imagine this is a place where a customer may want to do it within tests. Can you add a new section that notes "if you do want to include it in tests" with the blurbs on how filenames are used?  The final product would show your idiomatic way, but also have these instructions in case someone wanted to do it with the tests approach.